src/core/server/saved_objects/migrations/actions/initialize_action.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import * as TaskEither from 'fp-ts/lib/TaskEither';
+import * as Either from 'fp-ts/lib/Either';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { ClusterGetSettingsResponse } from '@elastic/elasticsearch/lib/api/types';
+import { ElasticsearchClient } from '../../../elasticsearch';
+import {
+  catchRetryableEsClientErrors,
+  RetryableEsClientError,
+} from './catch_retryable_es_client_errors';
+
+import { FetchIndexResponse, fetchIndices } from './fetch_indices';
+
+const routingAllocationEnable = 'cluster.routing.allocation.enable';
+export interface ClusterRoutingAllocationEnabled {
+  clusterRoutingAllocationEnabled: boolean;
+}
+
+export interface InitActionParams {
+  client: ElasticsearchClient;
+  indices: string[];
+}
+
+export const initAction = ({
+  client,
+  indices,
+}: InitActionParams): TaskEither.TaskEither<
+  RetryableEsClientError,
+  { indices: FetchIndexResponse; clusterRoutingAllocationEnabled: boolean }
+> => {
+  const checkClusterRoutingAllocationEnabledTask: TaskEither.TaskEither<
+    RetryableEsClientError,
+    { clusterRoutingAllocationEnabled: boolean }
+  > = () => {
+    return client.cluster
+      .getSettings({
+        include_defaults: true,
+        flat_settings: true,
+      })
+      .then((settings) => {
+        const clusterRoutingAllocations: string[] =
+          settings?.transient?.[routingAllocationEnable] ??
```

Comment:
We're calling the API with `include_defaults` and those settings should be returned in the response. `transient` settings on the other hand might be undefined, hence making these optional in the check and falling back to an empty array.

Reply:
I don't know this API very well, but my understanding is that the defaults from `include_defaults` will always be the same, users can't change the defaults. If users can't change these defaults then I don't think it's useful to return them since they'll always be the same.

On 8.0.0 include_defaults=true returns about 1400 lines of JSON so I just want to make sure it's necessary to have it?

Reply:
You're absolutely right and 1400 lines of JSON is far too much!
We're only interested in what users may have changed up to the point of migrations running again, so yes, no need to include the defaults I guess.

Reply:
I'm not sure we should exclude defaults, considering that these are configurable in the `elasticsearch.yml` file. I'm not sure if they are allowlisted on cloud but even if they're not, self-hosted deployments could still set the routing to something other than `all`.

Reply:
Update: when we don't include defaults in the `getSettings` API call _and_ disable replica allocation in `elasticsearch.yml`, the integration test I added for the `initAction` times out! This means we're not catching _static_ configurations when we exclude these. To be absolutely certain we're not missing something, I don't think we should exclude the defaults from the API call.

Reply:
good find, so it seems `elasticsearch.yml` _sets_ the defaults